### PR TITLE
Bnb test fix for different hardwares

### PR DESCRIPTION
**Pull request by Titus-von-Koeller**
Just updating the acceptable generated text, as this is slightly different based on hardware from what I can tell. The first commit is based on what I observed on my dev VM with A10G and the second commit is based on what I saw [in the failing BNB integration pipeline](https://github.com/huggingface/peft/actions/runs/7925057159/job/21637597387#step:9:83).

In my eyes, both text strings seem similar enough. I already spoke about this with @younesbelkada regarding the first commit. The second seems a similar case.

### CODE REVIEW

File: tests/quantization/bnb/test_mixed_int8.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ class BaseMixedInt8Test(unittest.TestCase):
     input_text = "Hello my name is"
     EXPECTED_OUTPUTS = set()
     EXPECTED_OUTPUTS.add("Hello my name is John.\nI am a friend of the family.\n")
+    EXPECTED_OUTPUTS.add("Hello my name is John.\nI am a friend of your father.\n")
```
Comment (younesbelkada) [inline tests/quantization/bnb/test_mixed_int8.py:100]:
```suggestion
    # Expected values on a A10
    EXPECTED_OUTPUTS.add("Hello my name is John.\nI am a friend of your father.\n")
```

File: tests/quantization/bnb/test_mixed_int8.py [hunk 2]
```diff
@@ -847,6 +848,7 @@ class MixedInt8GPT2Test(MixedInt8Test):
     EXPECTED_OUTPUTS = set()
     EXPECTED_OUTPUTS.add("Hello my name is John Doe, and I'm a big fan of")
     EXPECTED_OUTPUTS.add("Hello my name is John Doe, and I'm a fan of the")
+    EXPECTED_OUTPUTS.add("Hello my name is John Doe, and I am a member of the")
```
Comment (younesbelkada) [inline tests/quantization/bnb/test_mixed_int8.py:851]:
```suggestion
    # Expected values on a A10
    EXPECTED_OUTPUTS.add("Hello my name is John Doe, and I am a member of the")
```

**Review by younesbelkada — APPROVED** [review]
Thank you very much for updating the tests so that they pass on a A10 !

**Review by amyeroberts — APPROVED** [review]
LGTM - thanks for updating!

**Comment by Titus-von-Koeller** [comment]
Could one of you please merge? I don't have permission to do so.

**Pull request merged**